### PR TITLE
(#61) Added Drug Dictionary Tests

### PR DESCRIPTION
**Pull request by alinai**
@KateMashkinaNIH Please review #61. Thanks!

### CODE REVIEW

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        cy.get(`div#NCI-dd-app-root ${tag}`).should('have.text', title);
+    }
+});
+
+Then('the introductory text appears', () => {
+    cy.get('#NCI-dd-app-root').find('p').should('be.visible').invoke('text').should('not.be.empty');
+});
+
+And('search bar is displayed below the introductory text', () => {
+    cy.get('form[data-testid="tid-search-container"]').should('be.visible');
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:19]: So this steps duplicates glossary step - the only difference is locator. How about you update locator in glossary step so we can reuse it? Hint (use div.search-box-container to focus on search box container)
That way this step wont be needed

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    cy.document().then(doc => {
+        expect(doc.querySelector('form[data-testid="tid-search-container"]').parentNode.previousElementSibling.tagName).to.be.eq('P');
+    });
+});
+
+Then('{string} option is displayed below the search bar', (browse) => {
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:25]: This step is exact copy of glossary - why copy code when this step already exist ?

File: cypress/integration/common/GlossaryAppIntegration.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+Then('the page is showing the expand results for letter {string}', (letter) => {
+  let results;
+
+  cy.url().then(url => {
+    if (url.includes('espanol'))
+      results = `resultados de: ${letter}`;
+    else
+      results = `results found for: ${letter}`;
+
+    cy.get("div[class='dictionary-list-container results'] h4").should('include.text', results);
```
Comment (KateMashkinaNIH) [inline cypress/integration/common/GlossaryAppIntegration.js:52]: if you change locator to div[class*="results"] h4 then this step can be reused for drug dictionary too

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    cy.document().then(doc => {
+        const browseOption = doc.querySelector('span.browse');
+        expect(browseOption.firstChild.wholeText).to.include(browse);
+        expect(browseOption.parentNode.previousSibling.className).to.be.eq('drug-search');
+    })
+});
+
+Then('the page is showing the expand results for the letter {string}', (letter) => {
+    let results;
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:34]: see comment below in glossary test - this step can be deleted after locator is updated there

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+And('user clicks on result number {int}', (resultNumber) => {
+    cy.get('div.results dfn').eq(resultNumber - 1).find('a').first().click();
+});
+
+Then('user is redirected to definition page and url contains {string}', (urlText) => {
+    cy.url().should('contain', urlText);
+});
+
+And('term title appears', () => {
+    cy.get('.dictionary-definiton__term-title').should('be.visible');
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:51]: h1.dictionary-definiton__term-title (add h1 to make sure tag is a title tag)

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        expect(expectedLinks.parentNode.className).to.be.eq('dictionary-definiton__term-definition');
+    })
+});
+
+When('user types {string} in the search bar', (textEntered) => {
+    cy.get('input[id="keywords"]').clear().type(textEntered);
+});
+
+Then('autosuggest appears', () => {
+    cy.get('div.ncids-autocomplete__menu.--terms[data-testid="tid-auto-suggest-options"]').should('be.visible');
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:72]: This locator should be this
div[data-testid="tid-auto-suggest-options"] will suffice

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+And('user clears the search box', () => {
+    cy.get('input[id="keywords"]').clear();
+});
+
+And('user clicks {string} button', (searchButton) => {
+    cy.get('#btnSearch').trigger('click', { followRedirect: false });
+    cy.wait(500);
+});
+
+
```
Comment (KateMashkinaNIH) [inline cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:105]: please remove empty lines after this (only one empty line is required after colon

**Review by KateMashkinaNIH — COMMENTED** [review]
Also, since you copied Glossary test file over to Common folder, you no longer need it in the Glossary folder (just leave beforeEach)

File: cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js (hunk 2 above)
Comment (alinai) [reply cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:25]: Drug Dictionary and Glossary has different locators. If I try to use generic locator div.search-box-container, it fails for definition page as definition page has locator div.search-box-container.bottom

Comment (alinai) [reply cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js:25]: Glossary folder does not have the beforeEach file

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**